Autopilot/TelemetryManager/Src/telemetryStateClasses.cpp

```diff
@@ -95,18 +116,43 @@ telemetryState& readFromPathMode::getInstance()
 
 void analyzeDataMode::execute(telemetryManager* telemetryMgr)
 {
-    //set the dataValid here
+    
+    telemetryMgr->dataValid = 1;
+    telemetryMgr->dataValid &= sizeof(&_PMData) == 19;
+    telemetryMgr->dataValid &= &_PMData.beginLanding != nullptr;
+    telemetryMgr->dataValid &= &_PMData.beginTakeoff != nullptr;
+    telemetryMgr->dataValid &= &_PMData.holdingAltitude != nullptr;
+    telemetryMgr->dataValid &= &_PMData.holdingTurnDirection != nullptr;
+    telemetryMgr->dataValid &= &_PMData.holdingTurnRadius != nullptr;
+    telemetryMgr->dataValid &= &_PMData.homebase != nullptr;
+    telemetryMgr->dataValid &= &_PMData.initializingHomeBase != nullptr;
+    telemetryMgr->dataValid &= &_PMData.modifyId != nullptr;
+    telemetryMgr->dataValid &= &_PMData.nextId != nullptr;
+    telemetryMgr->dataValid &= &_PMData.numWaypoints != nullptr;
+    telemetryMgr->dataValid &= &_PMData.prevId != nullptr;
+    telemetryMgr->dataValid &= &_PMData.stoppingAltitude != nullptr;
+    telemetryMgr->dataValid &= &_PMData.stoppingDirectionHeading != nullptr;
+    telemetryMgr->dataValid &= &_PMData.stoppingLatitude != nullptr;
+    telemetryMgr->dataValid &= &_PMData.stoppingLongitude != nullptr;
+    telemetryMgr->dataValid &= &_PMData.takeoffDirectionHeading != nullptr;
+    telemetryMgr->dataValid &= &_PMData.waypointModifyFlightPathCommand != nullptr;
+    telemetryMgr->dataValid &= &_PMData.waypointNextDirectionsCommand != nullptr;
+    telemetryMgr->dataValid &= &_PMData.waypoints != nullptr;
```

Comment:
_PMData.waypoints is a pointer and does not need the & sign since the variable stores the memory address.

Reply:
also fixed in 4ff2194